### PR TITLE
feat: add execution condition to steps

### PR DESCRIPTION
**Pull request by ytvnr**
https://github.com/gravitee-io/issues/issues/6629

### CODE REVIEW

File: jackson/src/main/java/io/gravitee/definition/jackson/datatype/api/deser/StepDeserializer.java [hunk 1]
```diff
@@ -41,6 +41,10 @@ public Step deserialize(JsonParser jp, DeserializationContext ctxt) throws IOExc
         step.setName(node.path("name").asText());
         step.setDescription(node.path("description").asText());
         step.setPolicy(node.get("policy").asText());
+        final JsonNode executionCondition = node.get("executionCondition");
```
Comment (brasseld) [inline jackson/src/main/java/io/gravitee/definition/jackson/datatype/api/deser/StepDeserializer.java:44]: Just `condition` should be sufficient, isn't it ?

File: jackson/src/main/java/io/gravitee/definition/jackson/datatype/api/ser/StepSerializer.java [hunk 2]
```diff
@@ -39,6 +39,9 @@ public void serialize(Step step, JsonGenerator jgen, SerializerProvider provider
         jgen.writeStringField("description", step.getDescription());
         jgen.writeBooleanField("enabled", step.isEnabled());
         jgen.writeStringField("policy", step.getPolicy());
+        if (step.getExecutionCondition() != null) {
+            jgen.writeStringField("executionCondition", step.getExecutionCondition());
```
Comment (brasseld) [inline jackson/src/main/java/io/gravitee/definition/jackson/datatype/api/ser/StepSerializer.java:43]: same here

File: model/src/main/java/io/gravitee/definition/model/flow/Step.java [hunk 3]
```diff
@@ -44,6 +44,9 @@
     @JsonProperty("enabled")
     private boolean enabled = true;
 
+    @JsonProperty("executionCondition")
```
Comment (brasseld) [inline model/src/main/java/io/gravitee/definition/model/flow/Step.java:47]: same here ;)

**Review by brasseld — CHANGES_REQUESTED** [review]

**Review by gcusnieux — APPROVED** [review]

**Review by marcambier — APPROVED** [review]

File: jackson/src/main/java/io/gravitee/definition/jackson/datatype/api/deser/StepDeserializer.java (hunk 1 above)
Comment (ytvnr) [reply jackson/src/main/java/io/gravitee/definition/jackson/datatype/api/deser/StepDeserializer.java]: done :)

**Pull request merged**